### PR TITLE
Make data displayed on a graph downloadable as a CSV file

### LINKED ISSUE #50: Allow users to download data from a graph
Similarly to the way PCEX does, users will want to download data from a graph. Support CSV downloading, maybe also XLS?

### PR DESCRIPTION
**Pull request by corviday**
Adds a GraphDownloadButton to each graph. This button receives the same data as the graph and formats it as a CSV for download, plus an extra object (`graphMetadata`) containing contextual metadata like the emissions scenario and 

It looks like more code than it is - the only functional changes to the `*Graph.js` files was to add the button; the only functional change to the `*Display` components was to create the metadata object. I did linting on every file I "touched", so it looks like there's a lot of changes, but most can be ignored.

New functional code is all in GraphDownloadButton.js. 

Resolves #50

### CODE REVIEW

**Comment by corviday** [comment]
Demo is [here](https://services.pacificclimate.org/dev/scip/app/).

Here is a CSV from each graph, currently they contain no model or experiment metadata.
 
[Mean_Long_Term_annual_peak_flow_date__RXI1P1.csv](https://github.com/pacificclimate/scip-frontend/files/12741217/Mean_Long_Term_annual_peak_flow_date__RXI1P1.csv)
[stream_flow_by_calendar_month.csv](https://github.com/pacificclimate/scip-frontend/files/12741218/stream_flow_by_calendar_month.csv)
[stream_flow_by_calendar_day.csv](https://github.com/pacificclimate/scip-frontend/files/12741219/stream_flow_by_calendar_day.csv)

**Comment by corviday** [comment]
After discussion with @rod-glover and @jameshiebert , we've decided to include metadata like model and experiment as a second table in the CSV.

**Comment by corviday** [comment]
This is now ready for review. Demo is [here](https://services.pacificclimate.org/dev/scip/app/). To see the new functionality, pick a watershed or conservation unit under the map (the first one on each list is valid), select one of the "indicator" tabs, and pick an indicator (variable) under ths tabs. The graph should load, including a  button to download graph contents.

File: src/components/GraphDownloadButton/GraphDownloadButton.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    const blank = '\n';
+
+    // make second table, which contains data
+    const header = csvRow(data[0].x, `${layout.title} (${data[0].text[0]})`);
+    const series = _.map(data, (s) => csvRow(s.y, s.name, 3));
+    return [noCommas(meta), blank, header, noCommas(series)];
+  }
+
+  function fileName() {
+    // convert everything that isn't a letter, number, plus, minus, or
```
Comment (rod-glover) [inline src/components/GraphDownloadButton/GraphDownloadButton.js:40]: Huh. I was kind of wondering why you frequently use functions with no args that compute a simple value like this. But I realized that it encapsulates the computation and documents it by the name of the function. Nice.

**Review by rod-glover — APPROVED** [review]
LGTM! Nice work. :+1: 

A couple of questions about the CSV content for yearly indicators:
- The TL cell in the data table is `Mean Long Term annual peak flow date: RXI1P1 (julian day)`.
- Is there a case when more than one run (rXi1p1) is present?
- What would the TL cell contain then?

**Comment by corviday** [comment]
> Is there a case when more than one run (rXi1p1) is present?

I have checked and there isn't, but there _should_ be. We have 8 CanESM runs! This is suspicious and needs looking into. At a guess, I hardcoded the graph to one run when our data set was smaller for some reason and need to undo that now.

**Pull request merged**